stlab/concurrency/future.hpp

```diff
@@ -1204,8 +1209,8 @@ struct make_when_any<void> {
 /**************************************************************************************************/
 
 template <typename E, typename F, typename T, typename... Ts>
-auto when_any(E executor, F f, future<T> arg, future<Ts>... args) {
-    return make_when_any<T>::make(std::move(executor), std::move(f), std::move(arg), std::move(args)...);
+auto when_any(E executor, F&& f, future<T> arg, future<Ts>... args) {
+    return make_when_any<T>::make(std::move(executor), std::forward<F>(f), std::move(arg), std::move(args)...);
 }
```

Comment:
I think the other when_any and when_all can get forwarding references on F && f. Or?

Reply:
Fixed.